File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int len = arr.length;
        int[] src = new int[len];
        src[0] = arr[0];
        int count = 1;
        for (int i = 1; i < len; i++)
            if (src[count - 1] != arr[i])
                src[count++] = arr[i];        
        int[] answer = new int [count];
        System.arraycopy(src, 0, answer, 0, count);
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    auto it = arr.begin();
    int pre = *(it++);
    answer.push_back(pre);
    for (auto it = arr.begin(); it != arr.end(); it++)
        if (pre != *it)
            answer.push_back(pre = *it);
    return answer;
}